### DISCUSSION TITLE
How do I fire-and-forget a HTTP request from an API endpoint?

### DISCUSSION THREAD

**Discussion by venkr** [vercel/next.js#68588] [Help]
### Summary

Let's say I have an expensive side-effect that takes 30s to run, I'd like to invoke on a handler. I know I don't want to block the user-path though, so I've put this in an external API.

However, this HTTP endpoint is still designed to return a status code after it succeeds. How do I make sure I successfully send a request, but don't block on its completion?

Example approach 1:
```
await fetch(expensiveFunction)
return NextResponse.json(200)
```
This is bad, since it blocks the response to the user for 30s.

Example approach 2:
```
fetch(expensiveFunction)
return NextResponse.json(200)
```
This fails, since as soon as we respond to the user, we don't run this expensive background function. We usually don't even get to submit our HTTP request.

The ideal approach combines both:
Example approach 3:
```
await fetchButWaitUntilSubmitOnly(expensiveFunction)
return NextResponse.json(200)
```
This would be awesome, ie: wait until the HTTP request is successfully submitted, but do _not_ wait until the endpoint responds.

Is there any way to do this in Next.js?

**Comment by Juneezee** [comment]
> this HTTP endpoint is still designed to return a status code after it succeeds. How do I make sure I successfully send a request, but don't block on its completion?

Once you return a HTTP response, you cannot modify its status code.

**Comment by 0xcny** [comment]
Have you found a solution to this?

**Comment by icyJoseph** [comment]
There's a new API for this, https://nextjs.org/docs/app/api-reference/functions/after
